turbopack/crates/turbopack-ecmascript/src/tree_shake/graph.rs

```diff
@@ -708,7 +743,32 @@ impl DepGraph {
                                 write_vars: used_ids.write,
                                 eventual_write_vars: captured_ids.write,
                                 var_decls: [default_var.to_id()].into_iter().collect(),
-                                content: ModuleItem::ModuleDecl(item.clone()),
+                                content: ModuleItem::Stmt(Stmt::Decl(Decl::Var(Box::new(
+                                    VarDecl {
+                                        span: DUMMY_SP,
+                                        kind: VarDeclKind::Const,
+                                        decls: vec![VarDeclarator {
+                                            span: DUMMY_SP,
+                                            name: default_var.clone().into(),
+                                            init: Some(match &export.decl {
+                                                DefaultDecl::Class(c) => ClassExpr {
+                                                    ident: default_var.clone().into(),
+                                                    class: c.class.clone(),
+                                                }
+                                                .into(),
+                                                DefaultDecl::Fn(f) => FnExpr {
+                                                    ident: default_var.clone().into(),
+                                                    function: f.function.clone(),
+                                                }
+                                                .into(),
```

Comment:
Changing a `export default function x() {}` to `const D = function x() {}` would change the hoisting of the function.

I think it should be a function declaration here to keep the same hoisting: `function D() {}`